Tools-Override/ConstructSharedFx.targets

```diff
@@ -0,0 +1,64 @@
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="GenerateDepsJson" AssemblyFile="$(BuildToolsTaskDir)CoreFx.Tools.dll"/>
+
+  <PropertyGroup>
+    <NewSharedFxVersion>9.9.9</NewSharedFxVersion>
+    <OldSharedFxVersion>1.0.0</OldSharedFxVersion>
+    <TestSharedFxPath>$(TestSharedFxDir)\shared\Microsoft.NETCore.App\$(NewSharedFxVersion)</TestSharedFxPath>
+    <TestSharedFxPathHostFxr>$(TestSharedFxDir)\host\fxr\$(NewSharedFxVersion)</TestSharedFxPathHostFxr>
+    <ToolsDotNetCliSharedFxPath>$(ToolsDir)dotnetcli\shared\Microsoft.NETCore.App\$(OldSharedFxVersion)</ToolsDotNetCliSharedFxPath>
+    <ConstructSharedFxSem>$(ObjDir)constructsharedfxcomplete.sem</ConstructSharedFxSem>
+  </PropertyGroup>
+  
+  <!-- This will generate a version of the dotnet cli that runs with the live
+       build of corefx using the other components gathered from the stable 
+       version consumed from the Tools directory
+       To debug any issues set COREHOST_TRACE=1 and rerun dotnet
+  -->  
+  <Target Name="ConstructTestingHost"
+          BeforeTargets="Build;BuildAllProjects"
+          Condition="!Exists('$(ConstructSharedFxSem)') OR '$(ReconstructSharedFx)' == 'true'" >
+          
+    <ItemGroup>
+      <!-- This will only copy over the related file -->
+      <HostFxrFile Condition="'$(OS)'=='Windows_NT'" Include="$(RuntimePath)hostfxr.dll" />
```

Comment:
Ideally you wouldn't need to do any of that filtering here and instead you should just copy *.* from what we restored as part of test-runtime.